### PR TITLE
fix(bug): avoid name collisions and conflicting ids

### PR DESCRIPTION
**Pull request by noelmcloughlin**
### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [x] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [x] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [x] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Describe the changes you're proposing

Avoid salt conflicting IDs

### Pillar / config required to test the proposed changes

N/A

### Debug log showing how the proposed changes work

    Detected conflicting IDs, SLS IDs need to be globally unique.
    The conflicting ID is 'salt-minion' and is found in SLS 'base:salt.standalone' and SLS 
        'base:salt.minion'

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [x] Updated the relevant test pillar.

### CODE REVIEW

**Comment by noelmcloughlin** [comment]
I fixed CI versions.

File: salt/standalone.sls [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% from "salt/map.jinja" import salt_settings with context %}
 
-salt-minion:
+salt-minion-standalone:
```
Comment (daks) [inline salt/standalone.sls]: Just to understand:
* what is the problem with the previous ID? was it conflicting with another ID in the same formula? or in another formula?
* How is the new ID a better choice?

Comment (noelmcloughlin) [reply salt/standalone.sls]: Hi @daks, The error happens if salt combines `minion.sls` and `standalone.sls`, i.e. merged SLS files with identical ID is not allowed.  I was seeing problem on this branch https://github.com/saltstack-formulas/salt-formula/pull/421

Comment (daks) [reply salt/standalone.sls]: OK, fine. 

Aren't those two states designed to be incompatible? I mean `standalone.sls` is just `minion.sls` with a specific configuration, if I understand the README correctly.

If those are really incompatible and we don't want people to use both, maybe have an error is good, or maybe doc must be clear about that.

Comment (noelmcloughlin) [reply salt/standalone.sls]: Yes, they are definitely incompatible logic but so are conflicting IDs... both could be addressed but I'll keep this PR for the salt error.

**Comment by myii** [comment]
Let's look at this after finalising #448 and getting that merged.

**Comment by myii** [comment]
Is this covered by #421 now?

**Comment by noelmcloughlin** [comment]
Ah okay, It was adopted there independently of this. thanks.

**Pull request closed without merging**